src/test/java/com/beust/jcommander/ParametersRulesTest.java

```diff
@@ -0,0 +1,35 @@
+package com.beust.jcommander;
+
+import static java.lang.Boolean.TRUE;
+
+import java.util.Map;
+
+import org.testng.annotations.Test;
+
+public class ParametersRulesTest {
+
+    @Parameters(rules = QuietAndVerboseAreMutualExclusive.class)
+    class Flags {
+        @Parameter(names = "--quiet", description = "Do not output anything")
+        boolean quiet;
+
+        @Parameter(names = "--verbose", description = "Output detailed information")
+        boolean verbose;
+    }
+
+    public static class QuietAndVerboseAreMutualExclusive implements IRule {
+        @Override
+        public void validate(Map<String, Object> parameters) throws ParameterException {
+            if (parameters.get("--quiet") == TRUE && parameters.get("--verbose") == TRUE)
+                throw new ParameterException("--quiet and --verbose are mutually exclusive");
+        }
+    }
+
+    @Test(expectedExceptions = ParameterException.class,
+          expectedExceptionsMessageRegExp = "--quiet and --verbose are mutually exclusive")
+    public void testParameters() throws Exception {
+        Object o = new Flags();
+        JCommander.newBuilder().addObject(o).build().parse("--quiet", "--verbose");
```

Comment:
Where do you specify that this `JCommander` object should use the `IRule` you defined?

Also, `IRule` seems a bit too generic, maybe `ParameterValidator` or something more specific?

Reply:
> Where do you specify that this `JCommander` object should use the `IRule` you defined?

This is found [in line 11](https://github.com/cbeust/jcommander/pull/542/files#diff-34d5317f21488040e1df754bb25cc5c9b0eef7822307b7ed1f0159d739dcde8aR11-R12).

> Also, `IRule` seems a bit too generic, maybe `ParameterValidator` or something more specific?

I do not share this concern, but on the other hand renaming it is ok for me. I would propose to use the *plural* form `IParametersValidator` (like I have just done it in 41ca247b6aedcb47bb796c023622ee0bfe8d0f44) to make clear that this interface will check *all* parameters at once.